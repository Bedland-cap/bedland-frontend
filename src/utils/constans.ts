export const HEADER_HEIGHT = 70;
export const FOOTER_HEIGHT = 52;
