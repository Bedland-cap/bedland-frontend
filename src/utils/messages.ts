const ERROR_MESSAGES = {
  resetPasswordPage: {
    notWithPasswordsRules:
      'Password is incorrect, please try again after reviewing the rules above',

    passwordsDoesNotMatchUp: 'Passwords do not match to each other',
  },
};

export default ERROR_MESSAGES;
