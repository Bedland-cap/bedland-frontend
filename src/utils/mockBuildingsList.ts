import { Building } from 'store/reducers/building/building_slice';

const mockBuildingsList: Building[] = [
  {
    id: '1',
    name: 'Berry',
    address: 'Berry Street 21',
    managerId: '1',
    floors: '5',
  },
  {
    id: '2',
    name: 'Miedziana',
    address: 'Miedziana Street 3',
    managerId: '1',
    floors: '7',
  },
  {
    id: '3',
    name: 'Seaport',
    address: 'Seaport Avenue 51',
    managerId: '2',
    floors: '5',
  },
  {
    id: '4',
    name: 'Desert',
    address: 'Desert Street 2X',
    managerId: '1',
    floors: '6',
  },
  {
    id: '5',
    name: 'Lakewood',
    address: 'Lakewood Avenue 57',
    managerId: '3',
    floors: '10',
  },
];

export default mockBuildingsList;
