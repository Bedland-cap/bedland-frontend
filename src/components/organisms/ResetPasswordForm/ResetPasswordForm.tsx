import { useForm, SubmitHandler, FieldValues } from 'react-hook-form';
import { useContext } from 'react';
import ErrorMessage from 'components/atoms/ErrorMessage/ErrorMessage';
import { useNavigate } from 'react-router-dom';
import InputWithLabel from 'components/molecules/InputWithLabel/InputWithLabel';
import Button from 'components/atoms/Button/Button';
import { ThemeContext } from 'theme/ThemeContext';
import AlertIcon from 'assets/AlertIcon.svg';
import { RegisterOptions } from 'components/atoms/Input/Input.types';
import ResetPasswordFormContainer from './ResetPasswordForm.styled';
import { errorMessageHandler, watcher } from './ResetPasswordForm.utils';

const ResetPasswordForm = () => {
  const { palette } = useContext(ThemeContext);
  const navigate = useNavigate();
  const {
    register,
    handleSubmit,
    watch,
    formState: { errors },
  } = useForm({ mode: 'onChange' });

  const enteredPasswordWatch = watch('password', '');

  const errorConditionWrongRules = Boolean(
    errors.password || errors.confirmPassword,
  );

  let errorConditionIfPasswordsMatchUp = false;

  const whichErrorMessageCondition = Boolean(
    errors.password || (errorConditionIfPasswordsMatchUp && errors.password),
  );

  const resetPasswordRegOptions: RegisterOptions = {
    validate: (value) => {
      errorConditionIfPasswordsMatchUp = watcher(enteredPasswordWatch, value);
      return errorConditionIfPasswordsMatchUp;
    },
  };

  const onSubmit: SubmitHandler<FieldValues> = (data: FieldValues) => {
    // eslint-disable-next-line no-console
    console.log(data); // logic with fetch to the server should be here
    navigate('/');
  };

  return (
    <ResetPasswordFormContainer onSubmit={handleSubmit(onSubmit)}>
      <InputWithLabel
        label='Your new password'
        input='password'
        placeholder='Enter your new password'
        type='password'
        register={register}
      />
      <div>
        <InputWithLabel
          label='Confirm the password'
          input='confirmPassword'
          placeholder='Confirm the password'
          type='password'
          register={register}
          registerOptions={resetPasswordRegOptions}
        />

        {errorConditionWrongRules ? (
          <ErrorMessage>
            <img
              src={AlertIcon}
              alt={errorMessageHandler(whichErrorMessageCondition)}
            />
            <div>{errorMessageHandler(whichErrorMessageCondition)}</div>
          </ErrorMessage>
        ) : null}
      </div>

      <Button
        backgroundColor={palette.secondaryDark}
        textColor={palette.light}
        type='submit'
        variant='primary'
      >
        Reset my password
      </Button>
    </ResetPasswordFormContainer>
  );
};

export default ResetPasswordForm;
