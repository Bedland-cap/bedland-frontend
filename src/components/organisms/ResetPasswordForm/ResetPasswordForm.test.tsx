import renderer from 'react-test-renderer';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';
import { act, fireEvent, render, screen } from '@testing-library/react';
import ThemeProvider from 'theme/ThemeContext';
import ERROR_MESSAGES from 'utils/constants/messages';
import ResetPasswordForm from './ResetPasswordForm';
import { errorMessageHandler, watcher } from './ResetPasswordForm.utils';

const mockUseNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as any),
  useNavigate: () => mockUseNavigate,
}));

describe('ResetPassword Form', () => {
  it('renders correctly', () => {
    const snapshot = renderer
      .create(
        <ThemeProvider>
          <BrowserRouter>
            <ResetPasswordForm />
          </BrowserRouter>
        </ThemeProvider>,
      )
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('The password is according to the rules and passwords match', async () => {
    let resetPasswordButton: any | null = null;
    let inputPassword: any | null = null;
    let inputConfirmPassword: any | null = null;

    await act(() => {
      render(
        <ThemeProvider>
          <MemoryRouter initialEntries={['/resetPassword']}>
            <ResetPasswordForm />
          </MemoryRouter>
        </ThemeProvider>,
      );
    });

    await act(async () => {
      resetPasswordButton = screen.getByRole('button');
      inputPassword = screen.getByPlaceholderText('Enter your new password');
      inputConfirmPassword = screen.getByPlaceholderText(
        'Confirm the password',
      );

      fireEvent.input(inputPassword, {
        target: {
          value: 'Aaaaaaaaa2.',
        },
      });

      fireEvent.input(inputConfirmPassword, {
        target: {
          value: 'Aaaaaaaaa2.',
        },
      });

      fireEvent.submit(resetPasswordButton as HTMLElement);
    });

    expect(mockUseNavigate).toHaveBeenCalled();
    expect(mockUseNavigate).toHaveBeenCalledTimes(1);
    expect(mockUseNavigate).toHaveBeenCalledWith('/');
  });

  it('The password is according to the rules and passwords dont match', async () => {
    let resetPasswordButton: any | null = null;
    let inputPassword: any | null = null;
    let inputConfirmPassword: any | null = null;

    await act(() => {
      render(
        <ThemeProvider>
          <MemoryRouter>
            <ResetPasswordForm />
          </MemoryRouter>
        </ThemeProvider>,
      );
    });

    await act(async () => {
      resetPasswordButton = screen.getByRole('button');
      inputPassword = screen.getByPlaceholderText('Enter your new password');
      inputConfirmPassword = screen.getByPlaceholderText(
        'Confirm the password',
      );

      fireEvent.input(inputPassword, {
        target: {
          value: 'Aaaaaaaaa2.',
        },
      });

      fireEvent.input(inputConfirmPassword, {
        target: {
          value: 'Aaaaaaaaa2',
        },
      });

      fireEvent.submit(resetPasswordButton as HTMLElement);
    });

    expect(
      await screen.getByText(
        ERROR_MESSAGES.resetPasswordPage.passwordsDoesNotMatchUp,
      ),
    ).not.toBeNull();
  });

  it('The password is not according to the rules and passwords match ', async () => {
    let resetPasswordButton: any | null = null;
    let inputPassword: any | null = null;
    let inputConfirmPassword: any | null = null;

    await act(() => {
      render(
        <ThemeProvider>
          <MemoryRouter>
            <ResetPasswordForm />
          </MemoryRouter>
        </ThemeProvider>,
      );
    });

    await act(async () => {
      resetPasswordButton = screen.getByRole('button');
      inputPassword = screen.getByPlaceholderText('Enter your new password');
      inputConfirmPassword = screen.getByPlaceholderText(
        'Confirm the password',
      );

      fireEvent.input(inputPassword, {
        target: {
          value: 'Aaaaaaaaa2',
        },
      });

      fireEvent.input(inputConfirmPassword, {
        target: {
          value: 'Aaaaaaaaa2',
        },
      });

      fireEvent.submit(resetPasswordButton as HTMLElement);
    });

    // expect(window.location.pathname).toBe('/');

    expect(
      await screen.getByText(
        ERROR_MESSAGES.resetPasswordPage.notWithPasswordsRules,
      ),
    ).not.toBeNull();
  });
  it('The password is not according to the rules and passwords dont match ', async () => {
    let resetPasswordButton: any | null = null;
    let inputPassword: any | null = null;
    let inputConfirmPassword: any | null = null;

    await act(() => {
      render(
        <ThemeProvider>
          <MemoryRouter>
            <ResetPasswordForm />
          </MemoryRouter>
        </ThemeProvider>,
      );
    });

    await act(async () => {
      resetPasswordButton = screen.getByRole('button');
      inputPassword = screen.getByPlaceholderText('Enter your new password');
      inputConfirmPassword = screen.getByPlaceholderText(
        'Confirm the password',
      );

      fireEvent.input(inputPassword, {
        target: {
          value: 'Aaaaaaaaa2',
        },
      });

      fireEvent.input(inputConfirmPassword, {
        target: {
          value: 'Aaaaaaaaa.2',
        },
      });

      fireEvent.submit(resetPasswordButton as HTMLElement);
    });

    // expect(window.location.pathname).toBe('/');

    expect(
      await screen.getByText(
        ERROR_MESSAGES.resetPasswordPage.notWithPasswordsRules,
      ),
    ).not.toBeNull();
  });
});
describe('ResetPassword Utils', () => {
  it('should return the correct error message when passwords do not match up', () => {
    const password = 'password123';
    const confirmPassword = 'password1234';

    const result = errorMessageHandler(watcher(password, confirmPassword));

    expect(result).toBe(
      ERROR_MESSAGES.resetPasswordPage.passwordsDoesNotMatchUp,
    );
  });

  it('should return the correct error message when passwords do not meet requirements', () => {
    const password = 'pass';
    const confirmPassword = 'pass';

    const result = errorMessageHandler(watcher(password, confirmPassword));

    expect(result).toBe(ERROR_MESSAGES.resetPasswordPage.notWithPasswordsRules);
  });
});
