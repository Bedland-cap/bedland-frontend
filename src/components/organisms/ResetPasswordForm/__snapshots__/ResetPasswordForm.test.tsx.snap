// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPassword Form renders correctly 1`] = `
.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 2.875rem;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
  width: 100%;
}

.c3 {
  border: 1px solid #909AFF;
  border-radius: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 2.875rem;
  padding: 0.625rem 1rem 0.625rem 1rem;
  width: 19.875rem;
}

.c3:disabled,
.c3:hover,
.c3:active,
.c3:focus {
  outline: none;
}

.c2 {
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 16;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 0.6875rem;
  width: 19.875rem;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #EF6517;
  border: 1px solid black;
  border-radius: 0.5rem;
  color: #FCFEFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 2.5rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0.625rem 1rem 0.625rem 1rem;
  width: 19.875rem;
}

.c5:disabled,
.c5:hover,
.c5:active,
.c5:focus {
  outline: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1rem;
  padding: 0 2.84375rem 0rem 2.84375rem;
  width: 100%;
}

<form
  className="c0"
  onSubmit={[Function]}
>
  <div
    className="c1"
  >
    <label
      className="c2"
      htmlFor="password"
      size="medium"
    >
      Your new password
    </label>
    <input
      className="c3"
      id="password"
      name="password"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Enter your new password"
      type="password"
    />
  </div>
  <div
    className="c1"
  >
    <label
      className="c2"
      htmlFor="confirmPassword"
      size="medium"
    >
      Confirm the password
    </label>
    <input
      className="c3"
      id="confirmPassword"
      name="confirmPassword"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="Confirm the password"
      type="password"
    />
  </div>
  <span
    className="c4"
  />
  <button
    className="c5"
    type="submit"
  >
    Reset my password
  </button>
</form>
`;
